Build fails on RISC-V

**GitHub is for bugs and features - not for support**
For support, please use the hashcat forums https://hashcat.net/forum/

**Check the FAQ**
Some items that might appear to be issues are not issues. Please review the hashcat FAQ https://hashcat.net/wiki/doku.php?id=frequently_asked_questions before submitting a bug report.

**Describe the bug**
A clear and concise description of what the bug is.

The `opt.c` file has some x86_64 intrinsic, which causes error on RISC-V.

```text
In file included from deps/phc-winner-argon2-20190702/_hashcat/opt.c:26,
                 from src/bridges/bridge_argon2id_reference.c:20:
deps/phc-winner-argon2-20190702/_hashcat/blake2/blamka-round-opt.h:64:22: error: unknown type name ‘__m128i’; did you mean ‘__u128’?
   64 | static BLAKE2_INLINE __m128i fBlaMka(__m128i x, __m128i y) {
      |                      ^~~~~~~
      |                      __u128
deps/phc-winner-argon2-20190702/_hashcat/blake2/blamka-round-opt.h:64:38: error: unknown type name ‘__m128i’; did you mean ‘__u128’?
   64 | static BLAKE2_INLINE __m128i fBlaMka(__m128i x, __m128i y) {
      |                                      ^~~~~~~
      |                                      __u128
deps/phc-winner-argon2-20190702/_hashcat/blake2/blamka-round-opt.h:64:49: error: unknown type name ‘__m128i’; did you mean ‘__u128’?
   64 | static BLAKE2_INLINE __m128i fBlaMka(__m128i x, __m128i y) {
      |                                                 ^~~~~~~
      |                                                 __u128
deps/phc-winner-argon2-20190702/_hashcat/opt.c:110:24: error: unknown type name ‘__m128i’; did you mean ‘__u128’?
  110 | static void fill_block(__m128i *state, const block *ref_block,
      |                        ^~~~~~~
      |                        __u128
deps/phc-winner-argon2-20190702/_hashcat/opt.c: In function ‘next_addresses’:
deps/phc-winner-argon2-20190702/_hashcat/opt.c:157:5: error: unknown type name ‘__m128i’; did you mean ‘__u128’?
  157 |     __m128i zero_block[ARGON2_OWORDS_IN_BLOCK];
      |     ^~~~~~~
      |     __u128
deps/phc-winner-argon2-20190702/_hashcat/opt.c:158:5: error: unknown type name ‘__m128i’; did you mean ‘__u128’?
  158 |     __m128i zero2_block[ARGON2_OWORDS_IN_BLOCK];
      |     ^~~~~~~
      |     __u128
deps/phc-winner-argon2-20190702/_hashcat/opt.c:168:5: error: implicit declaration of function ‘fill_block’ [-Wimplicit-function-declaration]
  168 |     fill_block(zero_block, input_block, address_block, 0);
      |     ^~~~~~~~~~
deps/phc-winner-argon2-20190702/_hashcat/opt.c: In function ‘fill_segment’:
deps/phc-winner-argon2-20190702/_hashcat/opt.c:186:5: error: unknown type name ‘__m128i’; did you mean ‘__u128’?
  186 |     __m128i state[ARGON2_OWORDS_IN_BLOCK];
      |     ^~~~~~~
      |     __u128
In file included from deps/scrypt-jane-master/code/scrypt-jane-romix.h:4,
                 from src/bridges/bridge_scrypt_jane.c:15:
deps/scrypt-jane-master/code/scrypt-jane-salsa.h: In function ‘scrypt_test_mix’:
deps/scrypt-jane-master/code/scrypt-jane-salsa.h:112:16: warning: unused variable ‘cpuflags’ [-Wunused-variable]
  112 |         size_t cpuflags = detect_cpu();
      |                ^~~~~~~~
make: *** [src/bridges/bridge_argon2id_reference.mk:27: bridges/bridge_argon2id_reference.so] Error 1
make: *** Waiting for unfinished jobs....
```

**To Reproduce**
Please provide us with all files required to reproduce the bug locally on our development systems. For instance: hash files, wordlists, rule files, ...

Build with [Arch Linux PKGBUILD](https://gitlab.archlinux.org/archlinux/packaging/packages/hashcat/-/blob/main/PKGBUILD?ref_type=heads) on RISC-V.

**Expected behavior**
A clear and concise description of what you expected to happen.

Build success.

**Hashcat version (please complete the following information):**
 - OS: [e.g. Linux] Linux
 - Distribution: [e.g. Ubuntu 18.04] Arch Linux RISC-V
 - Version: [e.g. 6.2.0] master HEAD

**Additional context**
Add any other context about the problem here. For instance, it was working with hashcat version X.X.X (also please post output from older versions).

--- a/src/cpu_features.c
+++ b/src/cpu_features.c
@@ -16,6 +16,16 @@ int cpu_supports_avx2 ()     { return 0; }
 int cpu_supports_avx512f ()  { return 0; }
 int cpu_supports_avx512vl () { return 0; }
 
+#elif defined(__riscv)
+
+// TODO: Support RVV
+int cpu_supports_sse2 ()     { return 0; }
+int cpu_supports_ssse3 ()    { return 0; }
+int cpu_supports_xop ()      { return 0; }
+int cpu_supports_avx2 ()     { return 0; }
+int cpu_supports_avx512f ()  { return 0; }
+int cpu_supports_avx512vl () { return 0; }
+
 #else
 static inline void cpuid (u32 leaf, u32 subleaf, u32 *eax, u32 *ebx, u32 *ecx, u32 *edx)
 {

